### DISCUSSION TITLE
What kind of HTTP version does current ESP32 support?

### DISCUSSION THREAD

**Discussion by blaticslm** [espressif/arduino-esp32#7530] [Q&A] [answered]
I am in a research project that somehow needs to transmit data quickly. UDP is fast but gonna losing some data, and TCP is stable but too slow. I just read about QUIC protocol in HTTP/3 that which can transmit faster and more stable. Which HTTP version does ESP32 use like HTTP/1.1 or HTTP/2? Does ESP32 also support HTTP/3?

**Comment by pedrominatel** [comment, marked as answer]
Hi,

Using Arduino-ESP32 APIs, only HTTP/1.1 is supported. ESP-IDF supports [HTTP/2](https://github.com/espressif/esp-idf/tree/master/examples/protocols/http2_request).

**Reply by blaticslm** [reply]
thanks！

**Comment by blaticslm** [comment]
For another question, I saw that now HTTP/3 has been out there for around 2 years. When ESP32 will be available for this new faster protocol?

**Comment by davidacm** [comment]
Although Http/3 is not supported yet, what about QUIC protocol?
 quic is better than tcp in many situations. I want to develop a project using quic. So, I'm interested to know if there are some libraries to add support to the quic protocol on ESP32?
Currently I'm not interested in http/3, but I would like to use quic. I'm working with UDP directly, but it forces me to implement a lot of features to avoid packet loss. TCP is not suitable for my project due to its features, but QUIC would save me a lot of work.

**Reply by blaticslm** [reply]
My project too is getting involved in issues like sending real-time data with TCP, which is a terrible nightmare. In this case, you should think to write a multithread program to collect those data into a buffer and send it.
